jobserver/models.py

```diff
@@ -108,44 +124,54 @@ def get_absolute_url(self):
         return reverse("job-detail", kwargs={"pk": self.pk})
 
     @property
-    def is_complete(self):
-        return self.status_code is None and self.completed_at
+    def is_completed(self):
+        if not self.is_finished:
+            return False
 
-    @property
-    def is_dependency_failed(self):
-        return self.status_code == 7
+        return self.status_code is None and self.completed_at
 
     @property
     def is_failed(self):
+        if not self.is_finished:
+            return False
+
+        if self.status_code is None:
+            return False
+
         dependency_not_finished = 6
         dependency_failed = 7
         dependency_running = 8
 
-        non_failure_status = self.status_code in [
+        non_failure_statuses = [
             dependency_not_finished,
             dependency_failed,
             dependency_running,
         ]
 
-        return self.status_code and not non_failure_status
+        return self.status_code not in non_failure_statuses
```

Comment:
If a dependency fails, there's an argument for calling `dependency_failed` a failed state; we certainly don't go on to run the job even if requested.

Reply:
That's a very good point.  From a UI perspective… if my Job isn't running because something up the tree failed and my Job _isn't going to run_… that's a failure I'd say.